agreement/player.go

```diff
@@ -58,6 +58,15 @@ type player struct {
 	// Pending holds the player's proposalTable, which stores proposals that
 	// must be verified after some vote has been verified.
 	Pending proposalTable
+
+	// the history of arrival times of the lowest credential from previous
+	// ronuds, used for calculating the filter timeout dynamically.
+	lowestCredentialArrivals *credentialArrivalHistory
```

Comment:
I agree with Chris, this should be a value, not a pointer. The nil value is only used on reset on period > 0 but the same functionality is achieved with `writePtr = 0; full = false`.

Reply:
sounds good. I changed this.